### PR TITLE
Add support for groups with device_class forwarding to binary_sensor …

### PR DESCRIPTION
**Pull request by twhittock**
…icons and display strings

## Proposed change

A separate PR is being submitted to core to allow specifying an optional device class for group. This PR extends the frontend to be aware of the device class, if present, and update the presentation of the group to match the device class it is emulating.

This is a companion to home-assistant/core#36324

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @twhittock, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;frontend#6084).

Once you do that we will be able to review and accept this pull request.

Thanks!

**Comment by twhittock** [comment]
This links to https://github.com/home-assistant/core/pull/36324 and https://github.com/home-assistant/home-assistant.io/pull/13637

File: src/common/entity/compute_state_display.ts [hunk 1]
```diff
@@ -55,14 +55,21 @@ export const computeStateDisplay = (
     return formatDateTime(date, language);
   }
 
+  let forwardedDomain = domain;
+  if (domain === "group" && stateObj.attributes.device_class) {
+    // Forward the localisation lookup to use binary sensor,
+    // as the device class is forwarded from there, too.
+    forwardedDomain = "binary_sensor";
```
Comment (bramkragten) [inline src/common/entity/compute_state_display.ts:62]: A group can be more than just `binary_sensors`, limiting this to `binary_sensor` seems wrong.

**Comment by bramkragten** [comment]
Is there also an architecture issue for this?

File: src/common/entity/compute_state_display.ts (hunk 1 above)
Comment (twhittock) [reply src/common/entity/compute_state_display.ts:62]: It's in addition to current behaviour. device_class is entirely optional. I didn't see anything else which can be 'on' or 'off' like a group and had a device_class worth implementing.

**Comment by twhittock** [comment]
No architecture issue, I'm not affiliated with this project at all, it was an annoyance I experienced as a user, and decided to contribute.

**Pull request closed without merging**